extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,417 @@
+   * @return {string} url of the placeholder
+   * @private
+   * */
+  constructPlaceholderSrc_() {
+    let src = null;
+    if (this.widgetId_) {
+      const url = this.base_ + this.widgetId_ + '/thumbnail';
+      src = addParamToUrl(url, 'amp', '1');
+    } else {
+      src = addParamsToUrl(this.base_ + 'api/thumbnail', {
+        'url': this.url_,
+        'key': this.key_,
+        'amp': '1',
+      });
+    }
+    return src;
+  }
+
+  /**
+   * Test component call for required params
+   * @return {null}
+   * @private
+   * */
+  parseAttributes_() {
+    userAssert(
+      this.element.getAttribute('data-id') ||
+        this.element.getAttribute('data-url'),
+      'Iframely requires either "data-id" or a pair of "data-url" and "data-key" parameters for <%s> %s',
+      TAG,
+      this.element
+    );
+    if (!this.widgetId_) {
+      if (this.url_) {
+        userAssert(
+          this.key_,
+          'Iframely data-key must also be set when you specify data-url parameter at <%s> %s',
+          TAG,
+          this.element
+        );
+      }
+      if (this.key_) {
+        userAssert(
+          this.url_,
+          'Iframely data-url must also be set when you specify data-key parameter at <%s> %s',
+          TAG,
+          this.element
+        );
+        userAssert(
+          16 < this.key_.length || this.key_.length > 256,
+          'Iframely data-key should be between 16 and 256 characters parameter at <%s> %s',
+          TAG,
+          this.element
+        );
+      }
+      if (this.key_ || this.url_) {
+        userAssert(
+          !this.widgetId_,
+          'Iframely data-id should not be set when there is already a pair of data-url and data-key for <%s> %s',
+          TAG,
+          this.element
+        );
+      }
+    }
+    if ((this.widgetId_ && this.url_) || (this.widgetId_ && this.key_)) {
+      userAssert(
+        !this.widgetId_,
+        'Only one way of setting either data-id or data-url and data-key supported for <%s> %s',
+        TAG,
+        this.element
+      );
+    }
+    if (this.widgetId_) {
+      this.src_ = addParamToUrl(this.base_ + this.widgetId_, 'amp', '1');
+    } else {
+      this.src_ = addParamsToUrl(this.base_ + 'api/iframe', {
+        'url': this.url_,
+        'key': this.key_,
+        'amp': '1',
+      });
+    }
+  }
+
+  /**
+   * Parse other data-* attributes and append them to API query url
+   * @return {object} of Iframely options
+   * @private
+   * */
+  parseOptions_() {
+    const options = {};
+    const exclude = [
+      'data-id',
+      'data-domain',
+      'data-key',
+      'data-url',
+      'data-img',
+    ];
+    let data = this.element
+      .getAttributeNames()
+      .filter(name => name.startsWith('data-'));
+    data = data.filter(name => !exclude.includes(name));
+    data.forEach(
+      item =>
+        (options[item.split('data-').pop()] = this.element.getAttribute(item))
+    );
+    return options;
```

Comment:
1. `Element.dataset`
2. `src/object.js`#`omit()`

```suggestion
    // FYI: These are camelCased from data-some-attr to someAttr, map if needed.
    return omit(this.element.dataset, ['id', 'domain', 'key', 'url', 'img']);
```